src/main/scala/viper/gobra/ast/internal/Program.scala

```diff
@@ -782,6 +776,33 @@ case class MultisetConversion(expr : Expr)(val info: Source.Parser.Info) extends
   }
 }
 
+/* ** Mathematical Map expressions */
+
+/**
+  * Represents a mathematical map literal "mmap[`keys`]`values` { k_0: e_0, ..., k_n: e_n }",
+  * where `keyExprs` constitutes the vector "k_0, ..., k_n" of members of `keys` type
+  * and `valExprs` constitutes the vector "e_0, ..., e_n" of members,
+  * which should all be of type `values`.
+  */
+case class MathMapLit(keys : Type, values : Type, entries : Map[Expr, Expr])(val info : Source.Parser.Info) extends CompositeLit {
```

Comment:
You are totally right. I changed it to a sequence of pairs. This observation also made me realize that Go does not define an evaluation order for components of composite literals, which invalidates some assumptions that we make in Gobra (#279).

Reply:
I also realized that Go treats repeated keys in maps as an error when they can be evaluated at compile time and I introduced a corresponding check. Furthermore, I introduced a dynamic check that throws an error when the runtime value of two expressions used as keys in a map literal are equal. This way, we circumvent the need to handle undefined behavior.

Reply:
We do not have to check whether constant keys overlap at compile time. We can deal with that at verification time. We do not have to mirror the Go compiler.

Reply:
For instance, in theory, we could do it like Verifast and do the type checks at verification time, as well. What we care about are runtime errors. I.e. if Go throws an exception if a map literal uses the same key twice, then we also have to throw an exception, but if the Go compiler just complain because it likes to complain a lot, then we do not have to care about that in my opinion.

Reply:
Sure, that sounds reasonable but it is also good that Gobra fails fast when trivial properties are not met instead of possibly waiting a long time for verification to terminate with an error. I will keep that in mind in the future. Nonetheless, I already added the static checks to Gobra and it seems unecessary to remove them.

Reply:
It's fine to keep it like that. I second @Felalolf that we could save us some implementation work if we have to do a check at verification time anyways to not implement it statically as well